### PR TITLE
[new-lair-9] sqlite ipc lair-keystore server

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/lair/pull/67

[Video Walkthrough (3:14)](https://www.youtube.com/watch?v=aLqHuVTxKqI)

- fills out the needed apis on the sqlite store to be a valid LairStore
- ties together the pid_file checking and sqlite store to make a valid Lair Ipc Server implementation

### CODE REVIEW

**Review by maackle — APPROVED** [review]
High-level understanding achieved

File: crates/lair_keystore/src/store_sqlite.rs [hunk 1]
```diff
@@ -218,6 +304,168 @@ impl SqlPool {
...
+            let bytes = entry.encode()?;
+            let mut write = this.write().await;
+            if let Some(seed_info) = seed_info {
+                write
+                    .transaction(move |txn| {
+                        // if the entry type has seed info
+                        // add the extra columns so we can look up
+                        // the entry by public keys
+                        txn.execute_optional(
+                            r#"INSERT INTO lair_keystore (
```
Comment (thedavidmeister) [inline crates/lair_keystore/src/store_sqlite.rs:382]: should we have SQL in a separate file like we did with the scheduler?

File: crates/lair_keystore/src/store_sqlite.rs [hunk 2]
```diff
@@ -271,11 +519,40 @@ mod tests {
 
     #[tokio::test(flavor = "multi_thread")]
     async fn sqlite_sanity() {
+        // this is just a sanity / smoke test
+        // this is excersized better in the full server_test.rs
```
Comment (thedavidmeister) [inline crates/lair_keystore/src/store_sqlite.rs:523]: exercised

File: crates/lair_keystore/src/store_sqlite.rs [hunk 3]
```diff
@@ -44,11 +70,13 @@ impl Drop for SqlCon {
 }
 
 impl SqlCon {
-    pub fn last_insert_rowid(&self) -> i64 {
+    /*
+    fn last_insert_rowid(&self) -> i64 {
```
Comment (thedavidmeister) [inline crates/lair_keystore/src/store_sqlite.rs:74]: delet?

**Review by thedavidmeister — APPROVED** [review]

File: crates/lair_keystore/src/store_sqlite.rs (hunk 1 above)
Comment (neonphog) [reply crates/lair_keystore/src/store_sqlite.rs]: :+1: done

File: crates/lair_keystore/src/store_sqlite.rs (hunk 2 above)
Comment (neonphog) [reply crates/lair_keystore/src/store_sqlite.rs:478]:
```suggestion
        // this is exercised better in the full server_test.rs
```

**Pull request merged**